File: training/csharp/src/De.Sots.CellarsAndCentaurs/Domain/Model/Centaur.cs
namespace De.Sots.CellarsAndCentaurs.Domain.Model;

public sealed class Centaur : Creature
{
    private const int GALLOP_BONUS = 2;

    public Centaur(CreatureId id)
        : base(id, CreatureType.Monstrosity)
    {
        Speeds[SpeedType.Walking] = Speed.Of(50);
    }

    public override int Initiative() => base.Initiative() + GALLOP_BONUS;
}

File: sample-projects/ruby/lib/de/sots/cellars_and_centaurs/domain/model/centaur.rb
# frozen_string_literal: true

require_relative 'creature'
require_relative 'creature_type'
require_relative 'speed'
require_relative 'speed_type'

module De::Sots::CellarsAndCentaurs::Domain::Model
  class Centaur < Creature
    GALLOP = Speed.new(50)

    def initialize(id)
      super(id, CreatureType::MONSTROSITY)
      speeds[SpeedType::WALKING] = GALLOP
    end

    def initiative
      super + 2
    end
  end
end
